src/composables/useFlawCommentsModel.ts

```diff
@@ -0,0 +1,107 @@
+import { computed, ref, type Ref } from 'vue';
+import { getJiraComments, postJiraComment } from '@/services/JiraService';
+import type { ZodFlawCommentType, ZodFlawType } from '@/types/zodFlaw';
+import { postFlawComment } from '@/services/FlawService';
+
+export const SYSTEM_EMAIL = 'bugzilla@redhat.com';
+
+export function useFlawCommentsModel(flaw: Ref<ZodFlawType>, isSaving: Ref<boolean>, afterSaveSuccess: () => void) {
+  const internalCommentsAvailable = ref(false);
+  const isLoadingInternalComments = ref(false);
+  const internalComments = ref<ZodFlawCommentType[]>([]);
+
+  const publicComments = computed<ZodFlawCommentType[]>(() => flaw.value.comments.filter(filterFunctions.public));
+  const privateComments = computed<ZodFlawCommentType[]>(() => flaw.value.comments.filter(filterFunctions.private));
+  const systemComments = computed<ZodFlawCommentType[]>(() => flaw.value.comments.filter(filterFunctions.system));
+
+  type OsidbCommentFilter = 'public' | 'private' | 'system';
+  type OsidbCommentFilterFunctions = Record<OsidbCommentFilter, (comment: ZodFlawCommentType) => boolean>;
+
+  const filterFunctions: OsidbCommentFilterFunctions = {
+    public: (comment: any) => !comment.is_private,
+    private: (comment: any) => comment.is_private && comment.creator !== SYSTEM_EMAIL,
+    system: (comment: any) => comment.creator === SYSTEM_EMAIL,
+  };
+
+  function loadInternalComments() {
+    isLoadingInternalComments.value = true;
+
+    if (!flaw.value.task_key) {
+      internalCommentsAvailable.value = false;
+      isLoadingInternalComments.value = false;
+      return;
+    }
+
+    getJiraComments(flaw.value.task_key)
+      .then((res) => {
+        if (res.response.ok) {
+          internalCommentsAvailable.value = true;
+        }
+        if (res.data.comments.length >= 0) {
+          internalComments.value = parseJiraComments(res.data.comments);
+        }
+      })
+      .catch((res) => {
+        if (res?.response?.status === 404) {
+          console.error('Error loading internal Jira comments: Task id not found.');
+        } else {
+          console.error('Error loading internal Jira comments');
+        }
+        internalCommentsAvailable.value = false;
+      })
+      .finally(() => {
+        isLoadingInternalComments.value = false;
+      });
+  }
+
+  function addFlawComment(comment: string, creator: string, type: string) {
+    if (type !== 'Internal') {
```

Comment:
We have defined the comment types `Intenal, Public, Private, System` multiple times each time with different casing (`Public` vs `public`), I think it makes sense to have them in a constant and use it everywhere (like the enum in `FlawForm`

```typescript
const enum COMMENT_TYPES {
  PUBLIC = "Public",
  INTERNAL= "Internal",
  PRIVATE = "Private", 
  SYSTEM = "System"
}
```
This also provides a way to automatically calculate the type on https://github.com/RedHatProductSecurity/osim/blob/b1336d29423d1301f34abea96785053f1e64d216/src/composables/useFlawCommentsModel.ts#L17

```typescript
type OsimCommentFilter = Lowercase<Exclude<keyof typeof COMMENT_TYPES, "INTERNAL">>
//		^ = "public" | "private" | "system"
```

<details>
<summary>Always prefer constant enums</summary>

On typescript, when you use an enum
```typescript
enum Sample {
  VALUE1
  VALUE2
}

console.log(Sample.VALUE1);
```
it will create a runtime function for checking the values:

```js
var Sample;
(function (Sample) {
    Sample[Sample["VALUE1"] = 0] = "VALUE1";
    Sample[Sample["VALUE2"] = 1] = "VALUE2";
})(Sample || (Sample = {}));

console.log(Sample.VALUE1);
```

but if the enum is declared as constant (`const enum Sample { ... }`) it will just replace the usage with the constant value on build:
```js
console.log(0 /* Sample.VALUE1 */);
```
See the example in [Typescript Playground](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBApmArgWxgYQPIFksFEByAKgPqECaACrgMrGb7WEwDeAsAFAwwUCqAQgBkAkmhgBeGACIKiAEYAbAJbBJAGg5chRXACV8AQQETJQsFDgAnMAEN5ajdx1CAavsK5xUihcUA3a+ZqMA7UZIy4WJ6S1ACe0HDIkhwAvhwcCCjo2HhEpJQ0LA68giJRMgrK9pwwWu56hsam5la2VVwUTq7uZT7+garB1aHhkcax8YkpaewA9DMwoJBQ1mYw-vKIcDDgMLKIivIAJhyLECDycAB08iAA5gAUmDgEJORUtPSMl7W6BgIAlNM5jALIgzIpkFsAGZg4BQRTgE7gM4Xa53R7ZF55d7fbT1AEcIA)
</details>